Hmm, not as trivial as simply:

```diff
-  void Print(FGLogging& out);
+  void Print(FGLogging& out = FGLogging(LogLevel::STDOUT));
```

So, the change above compiled fine, with no warnings and ran fine locally with MSVC, however the non-MSVC compilers during the CI produced a compiler error.

```sh
FGTable.h:414:31: error: cannot bind non-const lvalue reference of type ‘JSBSim::FGLogging&’ to an rvalue of type ‘JSBSim::FGLogging’
  414 |   void Print(FGLogging& out = FGLogging(LogLevel::STDOUT));
```

https://www.stroustrup.com/C++11FAQ.html#rval

> In C++, non-const references can bind to lvalues and const references can bind to lvalues or rvalues, but there is nothing that can bind to a non-const rvalue. **That's to protect people from changing the values of temporaries that are destroyed before their new value can be used.**

In this particular case it wasn't an issue that the temporary was changed and destroyed, it flushed it's output when the temporary's destructor was called at the end of the the `Print()` method

Now it turns out that MSVC has a non-standard extension which allows a non-const reference to bind to an rvalue, and will display compiler warning C4239 when it's used, but the warning is only displayed if you enable warning level 4, which I didn't.

https://github.com/MicrosoftDocs/cpp-docs/blob/main/docs/error-messages/compiler-warnings/compiler-warning-level-4-c4239.md

> This type conversion is not allowed by the C++ standard, but it is permitted here as an extension. This warning is always followed by at least one line of explanation describing the language rule being violated.

So, I see 2 main options, both of which involve having 2 overloaded versions of `Print()`.

### Option 1

```c++
void Print()
{
   FGLogging out(LogLevel::STDOUT);
   Print(out);
}

void Print(FGLogging& out)
{
// Printing code
}
```

### Option 2

```c++
void Print(FGLogging&& out = FGLogging(LogLevel::STDOUT))  // rvalue reference
{
   Print(out);
}

void Print(FGLogging& out)
{
// Printing code
}
```

--- a/src/math/FGTable.cpp
+++ b/src/math/FGTable.cpp
@@ -749,6 +749,12 @@ FGTable& FGTable::operator<<(const double x)
 
 //%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+void FGTable::Print()
+{
+  FGLogging out(LogLevel::STDOUT);
+  Print(out);
+}
+
 void FGTable::Print(FGLogging& out)
 {
   out << std::setprecision(4);

--- a/src/math/FGTable.h
+++ b/src/math/FGTable.h
@@ -411,7 +411,8 @@ public:
 
   unsigned int GetNumRows() const {return nRows;}
 
-  void Print(FGLogging& out = FGLogging(LogLevel::STDOUT));
+  void Print();
+  void Print(FGLogging& out);
 
   std::string GetName(void) const {return Name;}
 

